tests/functional/test_inventories.py

```diff
@@ -14,30 +14,104 @@
+    return cluster_bundle.cluster_prototype().cluster_create(CLUSTER_NAME)
+
+
+def test_check_inventories_file(cluster: Cluster, provider: Provider, adcm_fs: ADCM, request: SubRequest):
     """Assert inventory file contents for the action"""
-    bundledir = utils.get_data_dir(__file__, 'cluster_inventory_tests')
-    cluster_bundle = sdk_client_fs.upload_from_fs(bundledir)
-    with allure.step('Create cluster'):
-        cluster_name = "dumMxpQA"
-        cluster = cluster_bundle.cluster_prototype().cluster_create(cluster_name)
-        cluster.service_add(name="zookeeper")
+    host = provider.host_create(fqdn=HOST_FQDN)
+    with allure.step('Configure cluster'):
+        service = cluster.service_add(name="zookeeper")
+        cluster.host_add(host)
+        component = service.component(name=GROUP_COMPONENT_NAME)
+        cluster.hostcomponent_set((host, component))
+    with allure.step(f'Create config group with {GROUP_COMPONENT_NAME} and change it'):
+        config_group = create_config_group_and_add_host('group_name', component, host)
+        config_group.config_set(CONFIG_TO_CHANGE)
+    with allure.step('Run actions on cluster'):
         run_cluster_action_and_assert_result(cluster, "set_multi_states")
         run_cluster_action_and_assert_result(cluster, "install")
     with allure.step('Get inventory file from container'):
         text = get_file_from_container(adcm_fs, '/adcm/data/run/2/', 'inventory.json')
-        inventory = json.loads(text.read().decode('utf8'))
-    with allure.step('Check inventory file'):
+        inventory_content = text.read().decode('utf8')
+        inventory = json.loads(inventory_content)
+        _attach_inventory_file(request, inventory_content, 'Actual content of inventory.json')
+    with allure.step('Get expected inventory file'):
         with open(utils.get_data_dir(__file__, 'cluster-inventory.json'), 'rb') as template:
-            expected = json.loads(template.read().decode('utf8'))
-            assert inventory == expected
+            expected_content = template.read().decode('utf8')
+            expected = json.loads(expected_content)
+        _attach_inventory_file(request, expected_content, 'Expected content of inventory.json')
+    with allure.step('Compare actual and expected config'):
+        assert (
+            inventory == expected
+        ), "Content of file inventory.json doesn't match expected. See attachments for more info."
+
+
+def _attach_inventory_file(request: SubRequest, inventory_content: str, name: str):
+    """Attach inventory file on top level of allure report"""
+    reporter = utils.allure_reporter(request.config)
+    if reporter:
+        test_result = reporter.get_test(uuid=None)
+        reporter.attach_data(
+            uuid=uuid4(),
+            body=inventory_content,
+            name=name,
+            attachment_type=allure.attachment_type.JSON,
+            parent_uuid=test_result.uuid,
+        )
+    else:
```

Comment:
if there is no reporter, `else` block do nothing

Reply:
removed